### PR TITLE
test:  adds tests for 'get_workflow_list'

### PR DESCRIPTION
**Pull request by danielboloc**
# Overview

Test the function `get_workflow_list` from `cloudos.clos`

Uses the `mock` and `responses` python packages to mock HTTP GET requests.

The test has the following functions each testing a different part:

- `test_get_workflow_list_correct_response` -> test if the function returns `requests.models.Response` and the `status_code` of the response is `200`

- `test_get_workflow_list_incorrect_response` -> test if the function raises the error `BadRequestException`

# Running the test

```console
python -m pytest -v tests/test_get_job_list.py
```

The response should be that all 2 tests have passed:

image

### CODE REVIEW

File: tests/test_get_workflow_list.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import mock
+import json
+import pytest
+import requests
+import responses
+import requests_mock
```
Comment (dapineyro) [inline tests/test_get_workflow_list.py:6]:
```suggestion
```
Not used

File: tests/test_get_workflow_list.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+from responses import matchers
+from cloudos.clos import Cloudos
+from cloudos.utils.errors import BadRequestException
+from tests.functions_for_pytest import load_json_file
+
+INPUT = "tests/test_data/process_job_list_initial_json.json"
+APIKEY = 'vnoiweur89u2ongs'
+CLOUDOS_URL = 'http://cloudos.lifebit.ai'
+WORKSPACE_ID = 'lv89ufc838sdig'
+
```
Comment (dapineyro) [inline tests/test_get_workflow_list.py:16]:
```suggestion


```
Two blank lines

File: tests/test_get_workflow_list.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            match=[matchers.query_param_matcher(params)],
+            status=200)
+    # start cloudOS service 
+    clos = Cloudos(apikey=APIKEY, cromwell_token=None, cloudos_url=CLOUDOS_URL)
+    # get mock response
+    response = clos.get_workflow_list(WORKSPACE_ID)
+    # check the response
+    assert response.status_code == 200
+    assert isinstance(response, requests.models.Response)
+
```
Comment (dapineyro) [inline tests/test_get_workflow_list.py:46]:
```suggestion


````
Two blank lines

File: tests/test_get_workflow_list.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    search_str = f"teamId={WORKSPACE_ID}&apikey={APIKEY}"
+    # mock GET method with the .json
+    responses.add(
+            responses.GET,
+            url=f"{CLOUDOS_URL}/api/v1/workflows?{search_str}",
+            body=create_json,
+            headers=header,
+            match=[matchers.query_param_matcher(params)],
+            status=200)
+    # start cloudOS service
```
Comment (dapineyro) [inline tests/test_get_workflow_list.py:39]:
```suggestion
    # start cloudOS service
```
Trailing whitespace

File: tests/test_get_workflow_list.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            url=f"{CLOUDOS_URL}/api/v1/workflows?{search_str}",
+            body=error_json,
+            headers=header,
+            match=[matchers.query_param_matcher(params)],
+            status=400)
+    # raise 400 error
+    with pytest.raises(BadRequestException) as error:
+        # check if it failed
+        clos = Cloudos(apikey=APIKEY, cromwell_token=None, cloudos_url=CLOUDOS_URL)
+        response = clos.get_workflow_list(WORKSPACE_ID)
```
Comment (dapineyro) [inline tests/test_get_workflow_list.py:75]:
```suggestion
        clos.get_workflow_list(WORKSPACE_ID)
```
Variable not used

**Review by dapineyro — CHANGES_REQUESTED** [review]
Great PR @danielboloc ! I just left some minor style comments. In addition, I'm a bit concerned about the use of `tests/test_data/process_job_list_initial_json.json` for mocking the server response. In this case, we are getting a "workflow list", not a job list. But for the scope of this testing function it's not relevant at all, so it's ok :+1:

Please, also remember to merge with `dev` and move the script to the `test_clos` folder.

**Comment by danielboloc** [comment]
@dapineyro thanks for the review! I have updated the input file to `tests/test_data/workflows/workflows.json` and applied the suggestions

**Review by dapineyro — APPROVED** [review]
LGTM! :+1: Please, before merging, update the branch to `dev` and be sure that the tests still run correctly

**Comment by danielboloc** [comment]
Thanks @dapineyro! Updated dev branch and tests still work

**Pull request merged**